### ISSUE TITLE
NM: Fix vote PDF parsing

### ISSUE THREAD

**Issue by mileswwatkins** [openstates/openstates#1276]
Right now, it appears that NM Senate vote PDFs aren't parsed correctly; [I've turned that off temporarily](https://github.com/openstates/openstates/pull/1188/commits/1207f3020f771baa35520c635cc507918e98c15b). And, FWIW, House vote PDFs only get their totals parsed, not their voting members.

**Comment by jonConerly** [comment]
I think I have a solution for this one that I can put a PR in for if you are not working on it already.

**Comment by mileswwatkins** [comment]
Fire away, @jonConerly!

**Comment by jonConerly** [comment]
I put in the PR for my change, this ended up being a can of worms.  I believe my method is less likely to break over time compared to the old method of pdf to text conversion; one primary reason for this is that it appears that the vote tally is being filled in by hand base on some metadata.  The plotted grid parser was also able to start parsing the house votes since their vote sheet is similar in concept, so I added that method in as well.

Added a tests folder as well to spot any future issues, any new pdfs added to the tests/testdate folder will be analyzed automatically if named correctly(*_senate.py and *_house.pdf).  nosetests was able to find and run the tests successfully.  My PR did not delete the old tests for the vote parsing regex so that might need to be cleaned up depending on what parts of this PR are accepted.
